docs/design/server_side_charts.md

```diff
@@ -0,0 +1,44 @@
+# Supporting charts on the server side
+As discussed in #367, the server side components manager and resourcifier have not yet been refactored, as of this writing, to work with [the new chart format](docs/design/chart_format.md). Presumably, expandybird should not be affected.
+
+## Context
+In discussions about the refactoring on the Slack channel, a number of potential requirements were discussed that could complicate matters, such as:
+* supporting multiple templating technologies within a single chart.
+* supporting multiple templating technologies within a single dependency graph.
```

Comment:
Re: dependency resolution, yes, a template would depend on a chart. Or, more precisely, a template would contain zero or more chart references, expressed in one of the 3 forms defined in the chart format document. In the general case, that becomes a URL that identifies the chart.

For there,miss up to the templating engine to chase the reference, retrieve the chart it points to, verify its signature, if it's signed, unzip it if necessary, unpack it if necessary and then extract its contents and use them appropriately.

As noted in the current text, DM does this today, with the caveat that it doesn't yet know about signing, zipping, archiving or the chart layout, which requires that the cotents live in a sub folder name templates. Today, it assumes that the URL points to the top level template file. References to other files required by the template are either relative to the template file, typically other files in the same folder, or URLs to be recursively resolved in the same manner.